card/config/initializers/01_core_extensions/hash.rb

```diff
@@ -52,6 +52,28 @@ def string_merge! other_hash, separator=" "
           old.is_a?(String) ? old + separator + new.to_s : new
         end
       end
+
+      # opposite of #dig
+      # eg. hash.bury(:a, :b, 7) will make sure hash[:a][:b] is eq
+      # @return [Hash] with arbitrary depth
+      def bury *array
+        validate_bury_array! array
+        key = array.shift
+        if array.size == 1
+          self[key] = array.first
+        else
+          self[key] ||= {}
+          self[key].bury(*array)
+        end
+        self
+      end
+
+      private
+
+      def validate_bury_array! array
+        return unless array.find { |a| a.is_a? Hash }
+        raise StandardError, "no hashes in bury arguments"
```

Comment:
`ArgumentError` seems to be a better fit

Reply:
good point